Sources/TSCclibc/process.c

```diff
@@ -13,7 +14,8 @@ int SPM_posix_spawn_file_actions_addchdir_np(posix_spawn_file_actions_t *restric
 #  if __GLIBC_PREREQ(2, 29)
     return posix_spawn_file_actions_addchdir_np(file_actions, path);
 #  else
-    return ENOSYS;
+    // Change working directory which child process will inherit
+    return chdir(path);
```

Comment:
please don't do this. This is very racy and affects the whole process.

Reply:
If you spawn two things at the same time this will fail horribly. `return ENOSYS` is correct here.

Reply:
Is there a way to make it non racy? Or alternative solution?

Reply:
With `posix_spawn`, the only way is `posix_spawn_file_actions_addchdir_np`.

You could use `fork()` + `execve()` and then you could `chdir` in the child process.

But if you don't want to rewrite to fork/exec and you don't have `posix_spawn_file_actions_addchdir_np`, then there's no way, no.

Reply:
does Foundation process solve this problem, or is `posix_spawn_file_actions_addchdir_np` used there as well?

Reply:
Foundation uses that function as well, which is the cause https://github.com/apple/swift/issues/59610, a fix for which had to be reverted.

Reply:
https://github.com/apple/swift-tools-support-core/pull/449/commits/0216fd2ca606737e1f803ea924911b53ac4fd4bc uses `fork` and `execve`